lib/pageflow/policies/entry_policy.rb

```diff
@@ -0,0 +1,71 @@
+module Pageflow
+  module Policies
+    class EntryPolicy
+      class Scope
+        attr_reader :user, :scope
+
+        def initialize(user, scope)
+          @user = user
+          @scope = scope
+        end
+
+        def resolve
+          scope.joins(sanitize_sql_array([
+            'LEFT OUTER JOIN pageflow_memberships ON ' \
+            'pageflow_memberships.user_id = :user_id AND ' \
+            'pageflow_memberships.entity_id = pageflow_entries.id AND ' \
+            'pageflow_memberships.entity_type = "Pageflow::Entry"',
+            user_id: user.id])).joins(sanitize_sql_array([
+              'LEFT OUTER JOIN pageflow_memberships as pageflow_memberships_2 ON ' \
+              'pageflow_memberships_2.user_id = :user_id AND ' \
+              'pageflow_memberships_2.entity_id = pageflow_entries.account_id AND ' \
+              'pageflow_memberships_2.entity_type = "Pageflow::Account"',
+              user_id: user.id])).where(
+                'pageflow_memberships.entity_id IS NOT NULL OR ' \
+                'pageflow_memberships_2.entity_id IS NOT NULL')
+        end
```

Comment:
This is still a bit hard to read. What do you think about refactoring it to something like

```
  scope
    .joins(memberships_for_entries(user))
    .joins(memberships_for_account_of_entries(user))
    .where(either_membership_is_present)
```